orderer/common/multichannel/registrar.go

```diff
@@ -188,19 +204,22 @@ func (r *Registrar) discoverLedgers(channelsWithJoinBlock map[string]*cb.Block)
 		if rl.Height() == 0 {
 			if _, ok := channelsWithJoinBlock[channelID]; !ok {
 				logger.Warnf("Channel '%s' has an empty ledger without a join-block, removing it", channelID)
-				if err := r.ledgerFactory.Remove(channelID); err != nil {
+				if err := r.ledgerFactory.Remove(channelID, r.removeChannelFromPendingRemovals); err != nil {
 					logger.Panicf("Ledger factory failed to remove empty ledger '%s', error: %s", channelID, err)
 				}
 			}
 		}
 	}
-
 	return r.ledgerFactory.ChannelIDs()
 }
 
 // initSystemChannel scan for and initialize the system channel, if it exists.
-func (r *Registrar) initSystemChannel(existingChannels []string) {
+func (r *Registrar) initSystemChannel(existingChannels []string, systemChannelID string) {
```

Comment:
The job of this method is to search for a system channel, so there really is no point in passing in that string